frame/contracts/proc-macro/src/lib.rs

```diff
@@ -140,3 +147,407 @@ fn format_default(field: &Ident) -> TokenStream {
 		&self.#field
 	}
 }
+
+/// Parsed environment definition.
+struct EnvDef {
+	host_funcs: Vec<HostFn>,
+}
+
+/// Parsed host function definition.
+struct HostFn {
+	item: syn::ItemFn,
+	module: String,
+	name: String,
+	returns: HostFnReturn,
+}
+
+enum HostFnReturn {
+	Unit,
+	U32,
+	ReturnCode,
+}
+
+/// Helper trait to convert a host function definition into its wasm signature.
+trait ToWasmSig {
```

Comment:
This trait looks redundant? Could make `to_wasm_sig` a simple inherent method by the looks of it